lib/KListWithOverflow.vue

```diff
@@ -101,6 +101,12 @@
       // avoids sharing it across multiple instances, ensuring each component has its own function.
       this.throttledSetOverflowItems = throttle(this.setOverflowItems, 100);
       this.$watch('elementWidth', this.throttledSetOverflowItems);
+
+      // Add resize observer to watch inner list items size changes
+      if (typeof window !== 'undefined' && window.ResizeObserver) {
```

Comment:
Do we need this and the resize observer for the element (provided by the responsive element mixin) or is this new resize observer all that is actually needed?

Reply:
Yes, we need both, one to watch the size of the wrapper of the list (which width is mainly set by the parent element), and this new Resize Observer to watch the internal changes.

Reply:
When unmounting the component, I think it'd be good to `.unobserve`. Or perhaps `.disconnect` would be better? https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver

Side note, I think I don't quite understand the previous discussion and haven't connect the dots with the example you provided yet. Would you try to briefly reformulate what's the situation when list items are being resized while the parent is not? That's rather for my understanding and learning. If it helps with the issue we're seeing, it's alright.

Reply:
Thank you! I totally missed to disconnect the observer.

And yes, the situation with both watchers are that we have two divs, the outher one is the root div of KListWithOverflow which takes 100% of the space available, it sets the limits, and the inner one is the div of the list which width depends on how many items fit inside the KListWithOverflow

![image](https://github.com/user-attachments/assets/8ce87aff-8c40-4907-856f-dcd79d037417)

The outher one will shrink/expand if the caller component is shrinked/expanded, but the inner one will only shrink/expand if the overflow items are recalculated (because of the wrapper size change and we are watching it) and there is one more/less item visible:

![image](https://github.com/user-attachments/assets/4251c4c9-f86f-4488-a5d5-f082bf0305f5)

But the thing is that the inner list items can also grow/shrink, and as we are not watching it, we cant recalculate the overflow items, and also because the inner list cant tell the outer one to expand/shrink to trigger the recalculation. So if some item is expanded we can get something like this:

![image](https://github.com/user-attachments/assets/673ba20d-282f-427f-be42-0c64a37d3d1a)


So thats why we need to watch the inner list. But now, if we just watch the inner list, we wouldnt be watching if the outher div expands, so we can reach a situation like this:

https://github.com/user-attachments/assets/e97e1ffe-20d1-45f3-8189-122ecf95e04c

So thats why we need to watch the outher div also.

Reply:
Ah tricky, thanks so much for explaining in detail